zha/application/discovery.py

```diff
@@ -306,6 +308,27 @@ def discover_quirks_v2_entities(self, device: Device) -> None:
                     )
                     cluster_handler.__dict__[zha_const.ZCL_INIT_ATTRS] = init_attrs
 
+                if (
+                    hasattr(entity_metadata, "attribute_name")
+                    and hasattr(entity_metadata, "reporting_config")
+                    and entity_metadata.reporting_config
+                ):
+                    reporting_config = tuple(
+                        filter(
+                            lambda cfg: cfg["attr"] != entity_metadata.attribute_name,
+                            cluster_handler.REPORT_CONFIG,
+                        )
+                    )
+                    reporting_config += (
+                        AttrReportConfig(
+                            attr=entity_metadata.attribute_name,
+                            config=astuple(entity_metadata.reporting_config),
+                        ),
+                    )
+
+                    cluster_handler.__dict__[zha_const.REPORT_CONFIG] = reporting_config
+
+                endpoint.claim_cluster_handlers([cluster_handler])
```

Comment:
Just noting (for myself) that claiming a cluster handler is what also makes ZHA configure and bind the cluster.